### PR TITLE
Kwxm/update plc spec mar 2021 (SCP-601)

### PR DESCRIPTION
**Pull request by kwxm**
This was just supposed to be a fix for a simple typo, but I realised that the specification was missing some not-so-recent changes and decided to update it a bit since there may be more people looking at it.  The main changes are
* It previously had the `(con 123)` syntax and I've changed that to `(con integer 123)` etc.
* I've also added the `bool` and `unit` types and the `ifThenElse` builtin and removed references to Scott booleans that we previously had.

Unfortunately, I now realise that there's a problem that it's quite difficult to fix: this version deals with saturated builtins and we now have unsaturated builtins, which has changed the syntax, typing rules, the semantics, and the workings of the machines.  There is a preliminary version of this in the new version of the specification that I've not had time to work on recently, but it would involve a major rewrite to add it to this document and I've already wasted too much time on the present PR.

I think there are two or three ways to proceed:

* Abandon these changes, which would leave us with a document that's quite misleading vis-a-vis the current state of the language.
* Add a paragraph saying how it's changed (in particular, the syntax) and that we'll give full details in a later version of the specification.
* Someone else could do it.

Any opinions?

Pre-submit checklist:
- Branch
    - [x] Commit sequence broadly makes sense
    - [x] Key commits have useful messages
    - [x] Relevant tickets are mentioned in commit messages
- PR
    - [x] Self-reviewed the diff
    - [x] Usexful pull request description
    - [x] Reviewer requested
- If you updated any cabal files or added Haskell packages:
    - [ ] `nix-shell shell.nix --run updateMaterialized` to update the materialized Nix files
    - [ ] Update `hie-*.yaml` files if needed
- If you changed any Haskell files:
    - [ ] `nix-shell shell.nix --run fix-stylish-haskell` to fix any formatting issues
- If you changed any Purescript files:
    - [ ] `nix-shell shell.nix --run fix-purty` to fix any formatting issues

Pre-merge checklist:
- [ ] Someone approved it
- [ ] Commits have useful messages
- [ ] Review clarifications made it into the code
- [ ] History is moderately tidy; or going to squash-merge

### CODE REVIEW

File: plutus-core-spec/figures/TypeSynthesis-Algorithmic.tex [hunk 1]
```diff
@@ -24,9 +24,9 @@
     \end{prooftree}
 
     \begin{prooftree}
-        \AxiomC{$cn$ has constant signature $\constsig{tcn}$ in Figure \ref{fig:constants}}
```
Comment (kwxm) [inline plutus-core-spec/figures/TypeSynthesis-Algorithmic.tex:27]: I don't think the previous version of this actually made sense.  Below the line it should probably have said `(con cn):(con tn)` or something.

**Comment by kwxm** [comment]
It seems that the CI problem form yesterday reappeared for some reason.  I've rebased again to try and fix it.

**Comment by michaelpj** [comment]
I think don't worry too much about it for now. Add a precautionary paragraph if you like and let's wait for V2!

**Comment by kwxm** [comment]
OK, I went with adding a warning and a brief explanation of how things have changed.

**Comment by michaelpj** [comment]
Kenneth, you've done something weird with Git again...

**Comment by effectfully** [comment]
Just tried to review the PR, but it includes irrelevant changes.

**Comment by kwxm** [comment]
Damn.  I'm not sure what happened there.  I rebased on master because of these VM failures, so maybe that's what caused it (I did do `git diff` before pushing and I didn't see anything out of the ordinary.  I'll see if I can fix it, maybe in a new branch.

**Comment by kwxm** [comment]
Closing this in favour of https://github.com/input-output-hk/plutus/pull/2947

**Pull request closed without merging**